### DISCUSSION TITLE
A question about the conda in linux

### DISCUSSION THREAD

**Discussion by lypeng7** [community/community#127197] [Programming Help]
### Select Topic Area

Question

### Body

I'm new to learning Linux.
I installed Ubuntu 22.04 on Windows 11. After installing Anaconda3, every time I run `source~/. bashc `, it resulted in errors.
```
(base) liao@LIAO-DESK:~$ source ~/.bashrc
-bash: eval: line 159: syntax error near unexpected token `('
-bash: eval: line 159: `export CONDA_PROMPT_MODIFIER='(base) ''
```
It didn't happen before I installed Anaconda3, and I cannot find line 159 in .bashrc?
Besides, when I run `conda activate wes` (wes is a env I create before), it shows
```
(base) liao@LIAO-DESK:~$ conda activate wes
-bash: syntax error near unexpected token `('
(wes) (base) liao@LIAO-DESK:~$
```
And I found that there is something strange about my Conda function, for example,
```
(wes) (base) liao@LIAO-DESK:~$ conda install samtools
Collecting package metadata (current_repodata.json): | WARNING conda.models.version:get_matcher(546): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.7.1.*, but conda is ignoring the .* and treating it as 1.7.1
done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 23.1.0
  latest version: 24.5.0

Please update conda by running

    $ conda update -n base -c defaults conda

Or to minimize the number of packages updated during conda update use

     conda install conda=24.5.0



## Package Plan ##

  environment location: /home/liao/anaconda3

  added / updated specs:
    - samtools


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2024.3.11  |       h06a4308_0         127 KB  https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
    certifi-2024.2.2           |  py310h06a4308_0         159 KB  https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
    openssl-1.1.1w             |       h7f8727e_0         3.7 MB  https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
    samtools-1.6               |       hb116620_7         514 KB  https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/bioconda
    ------------------------------------------------------------
                                           Total:         4.5 MB

The following NEW packages will be INSTALLED:

  samtools           anaconda/cloud/bioconda/linux-64::samtools-1.6-hb116620_7

The following packages will be UPDATED:

  ca-certificates    pkgs/main::ca-certificates-2023.01.10~ --> anaconda/pkgs/main::ca-certificates-2024.3.11-h06a4308_0

  certifi            pkgs/main::certifi-2022.12.7-py310h06~ --> anaconda/pkgs/main::certifi-2024.2.2-py310h06a4308_0
  openssl              pkgs/main::openssl-1.1.1t-h7f8727e_0 --> anaconda/pkgs/main::openssl-1.1.1w-h7f8727e_0


Proceed ([y]/n)? y


Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(wes) (base) liao@LIAO-DESK:~$ samtools --version
Command 'samtools' not found, but can be installed with:
sudo apt install samtools
```

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @lypeng7 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by lypeng7** [reply]
Thanks to you!

**Comment by mranv** [comment]
It looks like there are a few issues happening with your Anaconda installation and your `.bashrc` file. Let's break down and address each problem step by step.

### 1. Errors in `.bashrc`

The error messages indicate a syntax error around line 159 in your `.bashrc` file. Since you mentioned that you can't find line 159, it might be worth checking if there are any unintended line breaks or hidden characters causing this issue.

### 2. Issues with `conda activate`

The syntax error while running `conda activate` suggests that there might be an issue with how Anaconda is modifying your shell prompt.

### 3. Conda installation issues

The warning about the deprecated version specification and the failure to find installed packages indicate there might be an issue with your Conda environment or configuration.

### Steps to Resolve:

1. **Check and Clean `.bashrc`:**
   - Open your `.bashrc` file in a text editor:
     ```sh
     nano ~/.bashrc
     ```
   - Look for any lines that might be causing syntax errors, especially around line 159. Ensure there are no unintentional line breaks or special characters.
   - Remove or comment out the problematic lines if necessary.
   - Save and exit the editor.

2. **Reset Conda Initialization:**
   - Remove the existing Anaconda initialization in `.bashrc` if it's causing issues:
     ```sh
     conda init --reverse
     ```
   - Reinitialize Conda:
     ```sh
     conda init
     ```
   - This should add the correct initialization lines to your `.bashrc`.

3. **Update Conda:**
   - Make sure you have the latest version of Conda:
     ```sh
     conda update conda
     ```

4. **Verify Conda Environment:**
   - Create a new environment to see if the issue persists:
     ```sh
     conda create -n testenv python=3.10
     conda activate testenv
     ```
   - If this works without issues, it means your previous environment might have had configuration problems.

5. **Install Packages in the New Environment:**
   - Try installing `samtools` in the new environment:
     ```sh
     conda install samtools
     ```

6. **Check for System Conflicts:**
   - Ensure there are no conflicts between system-installed packages and Conda packages. For example, avoid installing `samtools` via `apt` if you are using the Conda version.

### Example `.bashrc` Initialization for Conda:

Make sure your `.bashrc` includes the correct Conda initialization lines. It should look something like this:

```sh
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/home/mranv/anaconda3/bin/conda' 'shell.bash' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/home/mranv/anaconda3/etc/profile.d/conda.sh" ]; then
        . "/home/mranv/anaconda3/etc/profile.d/conda.sh"
    else
        export PATH="/home/liao/anaconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<
```

### Restart Shell

After making these changes, restart your shell or source your `.bashrc` again:

```sh
source ~/.bashrc
```

This should resolve the issues you're facing with Anaconda on your Ubuntu installation within Windows 11. If the problems persist, consider reinstalling Anaconda to ensure a clean setup.
